qiskit_nature/mappers/second_quantization/bksf.py

```diff
@@ -0,0 +1,521 @@
+      str: One of 'number', 'excitation', 'coulomb_exchange',
+      'number_excitation', 'double_excitation'.
+
+    Raises:
+      ValueError: if the numbers of operators don't describe a one- or two-body term from
+      an electronic Hamiltonian.
+    """
+    if n_raise == 0 and n_lower == 0:
+        if n_number == 1:
+            return "number"
+        elif n_number == 2:
+            return "coulomb_exchange"
+        else:
+            raise ValueError("unexpected number of number operators")
+    elif n_raise == 1 and n_lower == 1:
+        if n_number == 1:
+            return "number_excitation"
+        elif n_number == 0:
+            return "excitation"
+        else:
+            raise ValueError("unexpected number of number operators")
+    elif n_raise == 2 and n_lower == 2:
+        return "double_excitation"
+    else:
+        raise ValueError("unexpected number of operators")
+
+
+def number_of_modes(fer_op: FermionicOp) -> int:
+    """Return the number of modes (including identities) in each term `fer_op`"""
+    return len(fer_op.to_list()[0][0])
+
+
+def operator_string(term: tuple) -> str:
+    """
+    Return the string describing the operators in the term extracted from a `FermionicOp`.
+    given by `term.
+    """
+    return term[0]
+
+
+def operator_coefficient(term: tuple) -> float:
+    """
+    Return the coefficient of the multi-mode operator term extracted from a `FermionicOp`.
+    """
+    return term[1]
+
+
+## TODO: We may want a lower-triangular matrix. This may be the cause of the minus sign error.
+def _get_adjacency_matrix(fer_op: FermionicOp) -> np.ndarray:
+    """
+    Return an adjacency matrix specifying the edges in the BKSF graph for the
+    operator `fer_op`.
+
+    The graph is undirected, so we choose to return the edges in the upper triangle.
+    (There are no self edges.). The lower triangle are all `False`.
+
+    Returns:
+          numpy.ndarray(dtype=bool): edge_matrix the adjacency matrix.
+    """
+    n_modes = number_of_modes(fer_op)
+    edge_matrix = np.zeros((n_modes, n_modes), dtype=bool)
+    for term in fer_op.to_list():
+        _add_edges_for_term(edge_matrix, operator_string(term))
+    return edge_matrix
+
+
+def _add_one_edge(edge_matrix, i: int, j: int) -> None:
+    """
+    Add an edge from lesser index to greater. This maintains the upper triangular structure.
+    """
+    if i < j:
+        edge_matrix[i, j] = True
+    elif j < i:
+        edge_matrix[j, i] = True
+    else:
+        raise ValueError("expecting i != j")
+
+
+def _add_edges_for_term(edge_matrix, term_str: str) -> None:
+    """
+    Add one, two, or no edges to `edge_matrix` as dictated by the operator `term_str`.
+    """
+    (n_number, n_raise, n_lower), facs = _unpack_term(term_str)
+    ttype = _interaction_type(n_number, n_raise, n_lower)
+    # For 'excitation' and 'number_excitation', create and edge between the `+` and `-`.
+    if ttype in ("excitation", "number_excitation"):
```

Comment:
ypThis usage strengthens @dlasecki's suggestion for an Enum.

Also, I typically see Python codes using an underscore-prefix to differentiate variables from builtin-names. I.e. `_type` rather than `ttype`

Reply:
I agree `ttype` -> `_type` is better.

Reply:
Both of these are done.